homeassistant/components/zwave_js/humidifier.py

```diff
@@ -0,0 +1,225 @@
+                value_property_key=setpoint_type,
+                add_to_watched_value_ids=True,
+            )
+            if setpoint_value:
+                self._setpoint_values[setpoint_type] = setpoint_value
+
+    def _setpoint_type_for_current_mode(self) -> HumidityControlSetpointType:
+        if self._current_mode:
+            return HUMIDITY_CONTROL_MODE_SETPOINT_MAP.get(
+                int(self._current_mode.value), []
+            )
+        return None
+
+    @property
+    def _setpoint_value(self) -> ZwaveValue | None:
+        """Optionally return a ZwaveValue for a setpoint."""
+
+        setpoint_types = self._setpoint_type_for_current_mode()
+        if len(setpoint_types) == 0:
+            return None
+
+        setpoint_type = setpoint_types[0]
+        if (
+            setpoint_type is None
+            or setpoint_type not in self._setpoint_values
+            or (val := self._setpoint_values[setpoint_type]) is None
+        ):
+            raise ValueError("Value requested is not available")
+
+        return val
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Set new target humidity."""
+        if self._current_mode:
+            if (
+                str(HumidityControlMode.AUTO.value)
+                in self._current_mode.metadata.states
+            ):
+                new_mode = HumidityControlMode.AUTO
+            elif (
+                str(HumidityControlMode.HUMIDIFY.value)
+                in self._current_mode.metadata.states
+            ):
+                new_mode = HumidityControlMode.HUMIDIFY
+            else:
+                new_mode = HumidityControlMode.DEHUMIDIFY
+            await self.info.node.async_set_value(self._current_mode, new_mode)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Set new target humidity."""
+        if self._current_mode:
+            await self.info.node.async_set_value(
+                self._current_mode, HumidityControlMode.OFF
+            )
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return True if entity is on."""
+        if self._current_mode is None:
+            return None
+        return cast(bool, int(self._current_mode.value) != HumidityControlMode.OFF)
+
+    @property
+    def target_humidity(self) -> int | None:
+        """Return the humidity we try to reach."""
+        if not self.is_on or self._setpoint_value is None:
+            return None
+        return int(self._setpoint_value.value)
+
+    @property
+    def mode(self) -> str | None:
+        """Return the current humidity control mode."""
+        if self._current_mode:
+            return cast(
+                str, self._current_mode.metadata.states[str(self._current_mode.value)]
+            )
+        return None
+
+    async def async_set_mode(self, mode: str) -> None:
+        """Set new mode."""
+
+        if self._current_mode is None:
+            return None
```

Comment:
Nothing is checking this return value.
```suggestion
            return
```

Reply:
can't apply this suggestion:

![image](https://user-images.githubusercontent.com/2053039/152703561-55dbd3f7-2c18-4741-9464-b9bb3236a8d1.png)